### ISSUE TITLE
Make `ucfirst` a method of `uppercase`

### ISSUE THREAD

**Issue by dpsanders** [JuliaLang/julia#21910]
E.g. with a keyword argument.

**Comment by stevengj** [comment]
Why?  In most languages these are distinct functions (e.g. ucfirst in Perl, capitalize in Python or Ruby, titleize in Rails, Title in Go)....

**Comment by stevengj** [comment]
(Arguably, `ucfirst` should convert the first character to titlecase, not to uppercase, however.)

**Comment by dpsanders** [comment]
The reasoning is to reduce API bloat, and because the name `ucfirst` does not tell me what the function does, since it uses an incomprehensible and non-standard abbreviation.

**Comment by dpsanders** [comment]
Actually, `capitalize` is not bad.

**Comment by stevengj** [comment]
Honestly, I don't see why we need `ucfirst` given `titlecase`.  I would think that in most circumstances where you'd use `ucfirst`, it would be for strings consisting of a single word, in which case `titlecase(s)` would be equivalent.

**Comment by stevengj** [comment]
I found 17 packages using `ucfirst`, and in all of them, as far as I can tell, they are applying it to single words where `titlecase` would work as well.

**Comment by dpsanders** [comment]
Closing, since `ucfirst` has now been moved to the `Unicode` stdlib package.

**Comment by rfourquet** [comment]
> Closing, since ucfirst has now been moved to the Unicode stdlib package.

But there is no separate issue tracker for stdlib packages, so couldn't this be left open?

**Comment by StefanKarpinski** [comment]
Yes, this is a good reason to move stdlib packages into their own repos at some point, but we're not there yet.
